File: bench/ack.js

function ack(m, n){
    if (m == 0) {
        return n+1;
    } else if (n == 0) {
        return ack((m - 1), 1);
    } else {
        var new_n = ack(m, (n - 1))
        return ack((m - 1), new_n);
    }
}
for (i = 0; i < 5; i++) {
    ack(3, 9);
}
console.time("cpu time");
for (i = 0; i < 10; i++) {
    ack(3, 9);
}
console.timeEnd("cpu time");

File: bench/ack.lua

function ack(m, n)
   if m == 0 then
      return n+1
   elseif n == 0 then
      return ack(m-1, 1)
   else
      return ack(m-1, ack(m, n-1))
   end
end

start_time=os.clock()
for i=0,10 do
   ack(3,9)
end
print("cpu time: ".. (os.clock()-start_time)*1000)
      
